test/search/test_search.cpp

```diff
@@ -543,15 +546,15 @@ void createIndices (std::vector<int>& indices, unsigned max_index)
 {
   // ~10% of the input cloud
   for (unsigned idx = 0; idx <= max_index; ++idx)
-    if (rand_uint () == 0)
+    if (rand_uint (rng) == 0)
       indices.push_back (idx);
    
-  boost::variate_generator< boost::mt19937, boost::uniform_int<> > rand_indices(boost::mt19937 (), boost::uniform_int<> (0, static_cast<int> (indices.size ()) - 1));
+  std::uniform_int_distribution<> rand_indices(0, static_cast<int> (indices.size ()) - 1);
```

Comment:
I don't see the point behind this `static_cast<int>`. Just remove it.

Reply:
`indices.size ()` is a `size_t`, so we have an unsigned type here. In case `size()` returns 0 we got with `static_cast<int> (indices.size ()) - 1` `-1` as value, during with without this cast we got `std::numeric_limits<size_t>::max()` => out of integer range => 0. Because this is test code it doesn't maybe matters. Just want to say we change behavior with it.